### DISCUSSION TITLE
API filter device by location name?

### DISCUSSION THREAD

**Discussion by llamafilm** [netbox-community/netbox#14558] [Q&A] [answered]
How can I filter devices by location name (either location.name or location.slug)?  Furthermore, how can I find out all the available filter fields?  I tried `/api/dcim/devices/?location=ct-sound-lab` but this returns _all_ devices as if no filter were applied.
I also tried `/api/dcim/devices/?location_slug=ct-sound-lab`, that does the same thing.

`/api/dcim/devices/?location_id=5` works as expected.

Here's an example of one device:
```json
            "location": {
                "id": 5,
                "url": "http://10.128.36.29/api/dcim/locations/5/",
                "display": "CT Sound Lab",
                "name": "CT Sound Lab",
                "slug": "ct-sound-lab",
                "_depth": 0
            },
```

**Comment by candlerb** [comment, marked as answer]
> `/api/dcim/devices/?location_id=5` works as expected.

The available filters are included in the machine-generated REST API documentation, linked from the bottom of your Netbox home page or at https://demo.netbox.dev/api/schema/swagger-ui/

Under `GET /api/dcim/devices/`:

![image](https://github.com/netbox-community/netbox/assets/44789/b3b25f55-3c06-4a0f-bb10-6dcda35e1007)

This shows that filtering on `location_id` is the only one available (`__n` is the negative variation)

You'll need to do a separate REST query to find the id of the location you're interested in. Or possibly graphQL can do it directly, I don't know.
